### PR TITLE
Disable MailtoDropdown on template loads

### PR DESCRIPTION
**Pull request by zner0L**
Currently, the production tests fail because of a race condition in the template load. To prevent that, we disbale the dropdown button while the template is loading.

### CODE REVIEW

File: src/Components/Generator/ActionButton.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ import MailtoDropdown from '../MailtoDropdown';
 
 export default class ActionButton extends Component {
     render(props) {
-        const enabled = props.transport_medium === 'email' ? !!props.letter : props.download_active;
+        const enabled =
+            props.transport_medium === 'email' ? !!props.letter && props.ready : props.download_active && props.ready;
```
Comment (baltpeter) [inline src/Components/Generator/ActionButton.js:9]: I personally would prefer this but that's not too important.

```suggestion
        const enabled = (props.transport_medium === 'email' ? !!props.letter : props.download_active) && props.ready;
```

**Review by baltpeter — CHANGES_REQUESTED** [review]
Unfortunately, now other tests are failing. :/

File: src/Components/RequestGeneratorBuilder.js [hunk 2]
```diff
@@ -377,11 +385,13 @@ export default class RequestGeneratorBuilder extends Component {
     handleCustomLetterTemplateChange = (e) => {
         const new_template = e.target.value;
         if (new_template !== 'no-template') {
+            this.setState({ ready: false });
             fetchTemplate(this.state.request.language, new_template, null, '').then((text) => {
                 this.changeRequest((req) => {
                     req.custom_data.content = text;
                     req.response_type = new_template;
                 });
+                this.setState({ ready: true });
```
Comment (baltpeter) [inline src/Components/RequestGeneratorBuilder.js:394]:
```suggestion
                this.setState({ ready: !!text });
```

File: src/Components/RequestGeneratorBuilder.js [hunk 3]
```diff
@@ -436,6 +447,7 @@ export default class RequestGeneratorBuilder extends Component {
                                     request_recipient_address: this.state.response_request.recipient,
                                 }).getText();
                             });
+                            this.setState({ ready: true });
```
Comment (baltpeter) [inline src/Components/RequestGeneratorBuilder.js:450]:
```suggestion
                            this.setState({ ready: !!text });
```

**Review by baltpeter — CHANGES_REQUESTED** [review]
I am a little worried about `this.setState({ ready: false });` possibly happening after `this.setState({ ready: true });` as `setState()` annoyingly isn't synchronous anymore. But since it's extremely unlikely that `fetch()` finishes before the state is flushed, I consider this unlikely enough for me not to care. Just for the record. :D

File: src/Components/RequestGeneratorBuilder.js [hunk 4]
```diff
@@ -133,7 +139,7 @@ export default class RequestGeneratorBuilder extends Component {
         // This is just a regular ol' request.
         else {
             await fetchTemplate(this.state.request.language, 'access').then((text) => {
-                this.setState({ template_text: text });
+                this.setState({ template_text: text, ready: true });
```
[the remark this answers is not in the document]
Comment (zner0L) [reply src/Components/RequestGeneratorBuilder.js:142]: Will you open an issue for this?

**Review by baltpeter — APPROVED** [review]
Thanks! Let's hope this works…

**Pull request merged**